torch/_dynamo/testing.py

```diff
@@ -245,6 +246,27 @@ def __call__(
         return gm.forward
 
 
+# Equivalent to backend="aot_eager", but also records graphs that
+# we can assert on
+class AOTEagerAndRecordGraphs:
+    def __init__(self) -> None:
+        self.graphs: List[torch.fx.GraphModule] = []
+
+    def __call__(
+        self, gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]
+    ) -> Callable[..., Any]:
+        def save_graph(gm: torch.fx.GraphModule, *args: Any, **kwargs: Any) -> Any:
+            self.graphs.append(gm)
+            return gm.forward
+
+        return aot_eager_decomp_partition(
```

Comment:
btw, can you modify the `aot_eager` backend instead? We're modifying `aot_eager_decomp_partition`. I know I sent you the link to the source code for aot_eager_decomp_partition, I actually meant to send you the source code for `aot_eager`

Reply:
NB: backend="aot_eager" is different from backend="aot_eager_decomp_partition" :P